care/facility/api/serializers/shifting.py

```diff
@@ -135,25 +175,44 @@ def update(self, instance, validated_data):
             validated_data.pop("patient")
 
         if "shifting_approving_facility" in validated_data:
-            shifting_approving_facility_external_id = validated_data.pop("shifting_approving_facility")["external_id"]
+            shifting_approving_facility_external_id = validated_data.pop(
+                "shifting_approving_facility"
+            )["external_id"]
             if shifting_approving_facility_external_id:
                 validated_data["shifting_approving_facility_id"] = Facility.objects.get(
                     external_id=shifting_approving_facility_external_id
                 ).id
 
         assigned = False
-        if "assigned_facility" in validated_data:
-            assigned_facility_external_id = validated_data.pop("assigned_facility")["external_id"]
+        if validated_data.get("assigned_facility"):
+            assigned_facility_external_id = validated_data.pop("assigned_facility")[
+                "external_id"
+            ]
             if assigned_facility_external_id:
                 validated_data["assigned_facility_id"] = Facility.objects.get(
                     external_id=assigned_facility_external_id
                 ).id
                 assigned = True
+        if validated_data.get("assigned_facility_external"):
```

Comment:
Not sure what is happening in this block

Reply:
Since we are assigning an external facility, there's is no data about it in the care platform, hence `assigned_facility_id` and `assigned_facility_object` cannot be set in this case. So, this block removes those. The only information we have about this external facility is its name. No `id` or `facility_object` is available to set. So they are set to `None`